pepper-apis/src/main/java/org/broadinstitute/ddp/DataDonationPlatform.java

```diff
@@ -598,4 +648,139 @@ private static void initSqlCommands(Config sqlConfig) {
         DBUtils.loadDaoSqlCommands(sqlConfig);
     }
 
+    /**
+     * RequestWrapper that uses a different header as the preferred source
+     * for the ip address that is returned by {@link HttpServletRequest#getRemoteAddr()}.
+     */
+    public static class SourceIPHeaderRequestWrapper extends HttpRequestWrapper {
+
+        String remoteAddr;
+
+        public SourceIPHeaderRequestWrapper(HttpServletRequest request, String preferredHeaderForIP) {
+            super(request);
+            String appEngineSourceIp = request.getHeader(preferredHeaderForIP);
+            remoteAddr = request.getRemoteAddr();
+            if (StringUtils.isNotBlank(appEngineSourceIp)) {
+                remoteAddr = appEngineSourceIp;
+            }
+        }
+
+        @Override
+        public String getRemoteAddr() {
+            return remoteAddr;
+        }
+    }
+
+    private static class AppEngineServerFactory implements EmbeddedServerFactory {
```

Comment:
This could probably be extracted out of DataDonationPlatform.

Reply:
Agreed--will find a better home.